### PR TITLE
EVG-2201: add logging when distro is underwater

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: scheduler/duration_based_host_allocator.go [hunk 1]
```diff
@@ -492,10 +494,37 @@ func (self *DurationBasedHostAllocator) numNewHostsForDistro(
...
+		"num_hosts": existingDistroHosts,
+	}
+
 	if !can {
+		underWaterAlert["action"] = []string{
+			"reduce workload;",
+			"add additional hosts to pool;",
+			"deactivate tasks;",
+		}
+		grip.AlertWhen(time.Duration(distroData.totalTasksDuration/float64(distro.PoolSize)) > 7*24*time.Hour,
```
Comment (john-m-liu) [inline scheduler/duration_based_host_allocator.go]: Let's put the value in a constant. Also this seems high - I feel like we want to alert well before hitting a week makespan per host

File: scheduler/duration_based_host_allocator.go [hunk 2]
```diff
@@ -492,10 +494,37 @@ func (self *DurationBasedHostAllocator) numNewHostsForDistro(
...
 	}
 
+	underWaterAlert["max_hosts"] = distro.PoolSize
+	underWaterAlert["actions"] = []string{
+		"provision additional hosts;",
+		"increase maximum pool size;",
+		"reduce workload;",
+		"deactivate tasks;",
+	}
+	grip.AlertWhen(time.Duration(distroData.totalTasksDuration/float64(distro.PoolSize)) > 24*time.Hour,
```
Comment (john-m-liu) [inline scheduler/duration_based_host_allocator.go]: Why not have a single alert for queue duration per host > 24 hours? It doesn't make sense to me to have different behavior depending on if we're at max hosts or not

Comment (john-m-liu) [inline scheduler/duration_based_host_allocator.go]: Should we also have alerter send an alert for a distro being underwater?

File: scheduler/duration_based_host_allocator.go (hunk 1 above)
Comment (tychoish) [reply scheduler/duration_based_host_allocator.go]: this is (I hope/suspect) for an entire host, not for a specific host. 

and for static hosts, many are underwater and under-provisioned to the point that I clear out tasks older than a week (on the theory that people don't care about them at that point.)

I'd like to be able to have an SLA of a <1 day for any task, but that doesn't seem realistic at this point.

File: scheduler/duration_based_host_allocator.go (hunk 2 above)
Comment (tychoish) [reply scheduler/duration_based_host_allocator.go]: the if is not "are we at max hosts" but rather are we static/not static (distro.PoolSize is 0 for statics.)

Comment (tychoish) [reply scheduler/duration_based_host_allocator.go]: I figured we'd set something up in splunk.

**Pull request merged**